pkg/controller/jobframework/reconciler.go

```diff
@@ -425,11 +426,20 @@ func (r *JobReconciler) startJob(ctx context.Context, job GenericJob, object cli
 }
 
 // stopJob will suspend the job, and also restore node affinity, reset job status if needed.
+// Returns whether any operation was done to stop the job or an error.
 func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, object client.Object, wl *kueue.Workload, eventMsg string) error {
 	info := getPodSetsInfoFromWorkload(wl)
 
 	if jws, implements := job.(JobWithCustomStop); implements {
-		return jws.Stop(ctx, r.client, info)
+		err := jws.Stop(ctx, r.client, info)
+		if err != nil {
+			r.record.Eventf(object, corev1.EventTypeNormal, "Stopped", eventMsg)
```

Comment:
I assume the event should be emitted if there is no error. 

But then, IIUC, the event is triggered on every `stopJob` invocation, which is on every reconciliation. Sounds excessive.

It seems we could emit the event only once when actual `.spec.suspend` is transitioned.

If we want to send the event at the end of the process of stopping (with reset of start time and node selectors) we may need to have a `Stopped` condition to mark that it was done without errors.

Reply:
sgtm